### DISCUSSION TITLE
Good examples of `vcpkg-cmake-wrapper.cmake`

### DISCUSSION THREAD

**Discussion by huangqinjin** [microsoft/vcpkg#42032] [Ideas]
`vcpkg-cmake-wrapper.cmake` is tricky to write. Need some examples for better understanding.

@dg0yt @Neumann-A

**Comment by dg0yt** [comment]
Start here: https://github.com/dg0yt/vcpkg/blob/1c66f2693096cc539042d01d8b2b4e0f07f4ae19/docs/maintainers/wrapper-guidelines.md 
(From https://github.com/microsoft/vcpkg/pull/19692)

**Reply by huangqinjin** [reply]
Kindly point out some concrete examples.

**Reply by dg0yt** [reply]
I mentioned tiff and libarchive in another post. Non-trivial, and maybe not perfect.

**Reply by huangqinjin** [reply]
Unfortunately, the wrapper of libarchive doesn't solve the most fundamental multi-config issue. It called to CMake's builtin find module which doesn't take care of mulit-config.
